src/components/DisplayTaxonName.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { Text } from "components/styledComponents";
+import _ from "lodash";
+import type { Node } from "react";
+import React from "react";
+
+type Props = {
+  item: Object,
+};
+
+const DisplayTaxonName = ( { item: { user, taxon } }: Props ): Node => {
+  if ( !taxon ) {
+    const text = "unknown";
+    return <Text numberOfLines={1}>{text}</Text>;
+  }
+  const taxonData = {};
+
+  taxonData.rank = taxon.rank;
+
+  if ( taxon.preferred_common_name ) {
```

Comment:
Could we add capitalization for the preferred common names? It could be pretty similar to what we're doing in [iNatModels.Taxon.titleCaseName](https://github.com/inaturalist/inaturalist/blob/1ff769e2df93b458e662200be080f55bd88f79ba/app/assets/javascripts/ang/models/taxon.js.erb#L169), and it probably makes sense for this to live in the Taxon model.

Reply:
Went ahead and added the two functions. I made some small adjustments like using a set for faster look ups, but the core logic is the same.

My understanding is, if part of the common name is more of an adjective and contains dashes, only the first letter will be capitalized, like in `Crown-of-thorns Sea Star`. But if the noun part of the common name has dashes, it will all be capitalized, like `Blue Night-Heron`

Please let me know if I've misunderstood the naming schema and I can make adjustments. Added a test case for the above as well.